### PR TITLE
add com.github.miguelmota.Cointop

### PR DESCRIPTION
**Pull request by miguelmota**
Adds [cointop](https://github.com/miguelmota/cointop) to flathub

Cointop is a simple terminal based application for tracking cryptocurrencies

Requires the home directory for storing configuration files, and network access for making HTTP requests

I'm new to flatpak; please let me know if it needs improvement

Thanks!

### CODE REVIEW

File: com.github.miguelmota.Cointop.json [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    {
+      "name": "cointop",
+      "buildsystem": "simple",
+      "build-commands": [
+        "install -D bin/linux/cointop /app/bin/cointop"
+      ],
+      "sources": [
+        {
+          "type": "git",
+          "url": "https://github.com/miguelmota/cointop.git"
```
Comment (Alexander-Wilms) [inline com.github.miguelmota.Cointop.json]: This should point to a specific release

**Comment by miguelmota** [comment]
@Alexander-Wilms added commit hash

**Comment by TingPing** [comment]
So this is a terminal application but since it is a TUI I think it will be fine. You need a destkop and appstream file though.

**Comment by TingPing** [comment]
> The first time you run cointop, it'll create a config file in:
> ~/.cointop/config

It also should be respecting the XDG spec: https://specifications.freedesktop.org/basedir-spec/basedir-spec-latest.html

(See XDG_CONFIG_HOME)

File: com.github.miguelmota.Cointop.json [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+  "runtime": "org.freedesktop.Platform",
+  "runtime-version": "1.6",
+  "sdk": "org.freedesktop.Sdk",
+  "command": "cointop",
+  "modules": [
+    {
+      "name": "cointop",
+      "buildsystem": "simple",
+      "build-commands": [
+        "install -D bin/linux/cointop /app/bin/cointop"
```
[the remark this answers is not in the document]
Comment (TingPing) [reply com.github.miguelmota.Cointop.json]: We already have the Go tools packaged. See this example: https://github.com/flathub/io.github.mki1967.mki3dgame/blob/039937faaf5e3df5eaad4ba13b17ea424d2caaef/io.github.mki1967.mki3dgame.json

File: com.github.miguelmota.Cointop.json [hunk 3]
```diff
@@ -0,0 +1,60 @@
+{
+  "app-id": "org.flatpak.Hello",
```
Comment (TingPing) [inline com.github.miguelmota.Cointop.json]: May have copied a bit too much.

File: com.github.miguelmota.Cointop.json [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+          "url": "https://github.com/miguelmota/cointop.git",
+          "commit": "70748c8e665e2608cdf88af91a180bcfd97dfcb1",
+          "dest": "go/src/github.com/miguelmota/cointop"
+        }
+      ]
+    }
+
+  ],
+  "finish-args": [
+    "--filesystem=host",
```
Comment (TingPing) [inline com.github.miguelmota.Cointop.json]: I don't think you need file system access at all. Again once you hopefully fix the config path issue discussed.

File: com.github.miguelmota.Cointop.json (hunk 3 above)
Comment (miguelmota) [reply com.github.miguelmota.Cointop.json]: fixed

File: com.github.miguelmota.Cointop.json (hunk 4 above)
Comment (miguelmota) [reply com.github.miguelmota.Cointop.json]: removed filesystem arg

File: flathub.json [hunk 5]
```diff
@@ -0,0 +1,3 @@
+{
+  "only-arches": ["x86_64", "i386"]
```
Comment (TingPing) [inline flathub.json]: I think these can be removed now?

Comment (miguelmota) [reply flathub.json]: done

**Comment by nedrichards** [comment]
bot, build com.github.miguelmota.Cointop

File: com.github.miguelmota.Cointop.json [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+        "install -D /app/go/src/github.com/miguelmota/cointop/x /app/bin/cointop",
+        "mkdir -p /app/share/icons/hicolor/64x64/apps",
+        "mkdir -p /app/share/icons/hicolor/128x128/apps",
+        "cp /app/go/src/github.com/miguelmota/cointop/assets/icon_64x64.png /app/share/icons/hicolor/64x64/apps/com.github.miguelmota.Cointop.png",
+        "cp /app/go/src/github.com/miguelmota/cointop/assets/icon_128x128.png /app/share/icons/hicolor/128x128/apps/com.github.miguelmota.Cointop.png"
+      ],
+      "sources": [
+        {
+          "type": "git",
+          "only-arches": ["x86_64", "i386"],
```
Comment (TingPing) [inline com.github.miguelmota.Cointop.json]: And here =)

Comment (miguelmota) [reply com.github.miguelmota.Cointop.json]: done

**Comment by TingPing** [comment]
Need a `.desktop` file still and might be good to go.

**Comment by miguelmota** [comment]
@TingPing can you point to an example `.desktop` file? 🙂

**Comment by miguelmota** [comment]
Never mind found the docs

**Comment by miguelmota** [comment]
hope it's good now

File: com.github.miguelmota.Cointop.json [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+        "cp -rpv go/* /app/go/",
+        "/app/bin/scripts/build.sh github.com/miguelmota/cointop",
+        "install -D /app/go/src/github.com/miguelmota/cointop/x /app/bin/cointop",
+        "mkdir -p /app/share/icons/hicolor/64x64/apps",
+        "mkdir -p /app/share/icons/hicolor/128x128/apps",
+        "mkdir -p /app/share/applications",
+        "cp /app/go/src/github.com/miguelmota/cointop/assets/icon_64x64.png /app/share/icons/hicolor/64x64/apps/com.github.miguelmota.Cointop.png",
+        "cp /app/go/src/github.com/miguelmota/cointop/assets/icon_128x128.png /app/share/icons/hicolor/128x128/apps/com.github.miguelmota.Cointop.png",
+        "cp /app/go/src/github.com/miguelmota/cointop/flathub/com.github.miguelmota.Cointop.desktop /app/share/applications/com.github.miguelmota.Cointop.desktop"
+      ],
```
Comment (TingPing) [inline com.github.miguelmota.Cointop.json]: Forgetting to install the appdata file into `/app/share/metainfo`

Comment (miguelmota) [reply com.github.miguelmota.Cointop.json]: added

File: com.github.miguelmota.Cointop.desktop [hunk 8]
```diff
@@ -0,0 +1,6 @@
+[Desktop Entry]
+Name=Cointop
+Exec=com.github.miguelmota.Cointop
```
Comment (TingPing) [inline com.github.miguelmota.Cointop.desktop]: So close, this needs to be the name of the binary. (Also please do test this by installing it and opening it from your launcher)

Comment (miguelmota) [reply com.github.miguelmota.Cointop.desktop]: @TingPing would `Exec=cointop` be correct? I'm testing it on ubuntu-server so I don't have a launcher

**Comment by TingPing** [comment]
bot, build com.github.miguelmota.Cointop

**Comment by miguelmota** [comment]
bot, build com.github.miguelmota.Cointop

**Comment by TingPing** [comment]
bot, build com.github.miguelmota.Cointop

**Comment by miguelmota** [comment]
👍 https://flathub.org/builds/#/builders/2/builds/1565

**Comment by TingPing** [comment]
Repository created: https://github.com/flathub/com.github.miguelmota.Cointop

**Comment by miguelmota** [comment]
This is awesome

**Pull request closed without merging**